tests/BuildScriptGenerator.Tests/Node/NodeVersionCheckerTest.cs

```diff
@@ -0,0 +1,59 @@
+﻿// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+using System.Linq;
+using System.Collections.Generic;
+using Microsoft.Oryx.BuildScriptGenerator.Node;
+using Xunit;
+
+namespace Microsoft.Oryx.BuildScriptGenerator.Tests.Node
+{
+    public class NodeVersionCheckerTest
+    {
+        [Fact]
+        public void Checker_Warns_WhenOutdatedVersionUsed()
+        {
+            // Arrange
+            var checker = new NodeVersionChecker();
+
+            // Act
+            var messages = checker.CheckToolVersions(
+                new Dictionary<string, string> { { NodeConstants.NodeJsName, "1.0.0" } });
+
+            // Assert
+            Assert.Single(messages);
+            Assert.Contains("outdated version of Node.js was used", messages.First().Content);
+        }
+
+        [Fact]
+        public void Checker_DoesNotWarn_WhenLtsVersionUsed()
+        {
+            // Arrange
+            var checker = new NodeVersionChecker();
+
+            // Act
+            var ltsVer = NodeScriptGeneratorOptionsSetup.NodeLtsVersion;
+            var messages = checker.CheckToolVersions(
+                new Dictionary<string, string> { { NodeConstants.NodeJsName, ltsVer } });
+
+            // Assert
+            Assert.Empty(messages);
+        }
+
+        [Fact]
+        public void Checker_DoesNotWarn_WhenCurrentVersionUsed()
+        {
+            // Arrange
+            var checker = new NodeVersionChecker();
+
+            // Act
+            var messages = checker.CheckToolVersions(
```

Comment:
Wouldn't this test fail in future (given how quick Node releases new versions 😄 )...You probably want to dynamically get the LTS version and increment it for this test to be resilient

Reply:
It will only fail when we change the value of `NodeScriptGeneratorOptionsSetup.NodeLtsVersion`.